crux_core/src/future.rs

```diff
@@ -1,3 +1,5 @@
+//! Async support for implementing capabilities
```

Comment:
This module isn't particularly useful from a documentation perspective...

Wonder if we want to re-jig it a bit.  Maybe make it a submodule of `capability` and just re-export `EffectFuture` from `capability`?  Not sure.

Reply:
The module isn't, but it contains one of the most important capability APIs. It'd be nice if we can make it part of `capability`.

Reply:
The function is defined in this file but it's part of `CapabilityContext` which isn't defined in this file, so rustdoc puts the docs for it elsewhere. 

Ofc the future module itself isn't public so doesn't appear in the documentation anyway.  So lets consider this comment resolved